runtime/compiler/optimizer/J9ValuePropagation.cpp

```diff
@@ -2808,17 +2810,17 @@ J9::ValuePropagation::transformVTObjectEqNeCompare(TR_OpaqueClassBlock *containi
       lhsOffsetNode->setIsInternalPointer(true);
       rhsOffsetNode->setIsInternalPointer(true);
 
-      if (isObjectEqualityCompare)
-         {
-         TR::Node *arraycmpNode = TR::Node::create(TR::arraycmp, 3, lhsOffsetNode, rhsOffsetNode, totalFieldSizeNode);
-         arraycmpNode->setSymbolReference(comp()->getSymRefTab()->findOrCreateArrayCmpSymbol());
-
-         TR::Node::recreateWithoutProperties(callNode, TR::ixor, 2, arraycmpNode, TR::Node::iconst(callNode, 1));
-         }
-      else
-         {
-         TR::Node::recreateWithoutProperties(callNode, TR::arraycmp, 3, lhsOffsetNode, rhsOffsetNode, totalFieldSizeNode, comp()->getSymRefTab()->findOrCreateArrayCmpSymbol());
-         }
+      // arraycmp returns 0, 1 or 2 - zero, for equality; one or two for inequality
+      // In order to yield a result in {0,1}, so transform object{Equality|Inequality}Comparison into
+      //
+      //   icmpeq
```

Comment:
To be consistent with the above example that uses `<objectInequalityComparison>`, should this use `icmpne` as well? Or the above example could be changed to `<objectEqualityComparison>`.

Reply:
Fixed in 5555dc7cdbe1aef429959c6471286298fc4efd4e